.github/workflows/update-latest-branch.yaml

```diff
@@ -0,0 +1,91 @@
+name: Update `latest` branch
+
+on:
+  workflow_call:
+    inputs:
+      branch:
+        type: string
+        required: true
+      dry_run:
+        type: boolean
+        required: false
+        default: false
+
+jobs:
+  update-latest-branch:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Update latest branch
+        id: update-latest-branch
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        env:
+          BRANCH: ${{ inputs.branch }}
+          DRY_RUN: ${{ inputs.dry_run }}
+        with:
+          script: |
+            const { BRANCH: currentBranch, DRY_RUN: dryRunStr } = process.env;
+            const dryRun = dryRunStr == "true";
+
+            // Get all branches
+            const branches = await github.paginate(github.rest.repos.listBranches, context.repo);
+
+            const tryUpdateBranch = async (to, from) => {
+              if (currentBranch != to) {
+                if (dryRun) {
+                  console.log(`Would have skipped branch "${to}" since it is not the current branch ("${currentBranch}")`);
+                } else {
+                  console.log(`Skipping branch "${to}" since it is not the current branch ("${currentBranch}")`);
+                }
+                return;
+              }
+
+              const ref = await github.rest.git.getRef({ ...context.repo, ref: `heads/${to}` });
+              const sha = ref.data.object.sha;
+
+              if (branches.some((b) => b.name == from)) {
+                if (dryRun) {
+                  console.log(`Would have updated branch "${from}" to point to branch "${to}", SHA ${sha}`);
+                } else {
+                  console.log(`Updating branch "${from}" to point to branch "${to}", SHA ${sha}`);
+                  await github.rest.git.updateRef({ ...context.repo, ref: `heads/${from}`, sha, force: true });
+                }
+              } else {
+                if (dryRun) {
+                  console.log(`Would have created branch "${from}" to point to branch "${to}", SHA ${sha}`);
+                } else {
+                  console.log(`Creating branch "${from}" to point to branch "${to}", SHA ${sha}`);
```

Comment:
I'm not sure if I want to let this bot create branches. If the bot creates a branch, it won't have the right protections. Perhaps we should require a manual step from ops to create the branch first, and set the right protections.